4-sinatra/forgot_password.rb

```diff
@@ -0,0 +1,19 @@
+require 'securerandom'
+
+puts "Podaj nowe hasło:"
+pass1 = gets.chomp
+puts "Potwierdz hasło:"
+pass2 = gets.chomp
+
+salt = SecureRandom.hex
+timestamp = Time.now.strftime('%s')
+
+hashed_password = Digest::SHA2.new(512).hexdigest(salt+"#"+pass1+"#"+timestamp)
+password = "#{salt}:#{hashed_password}:#{timestamp}"
+
+if pass1 == pass2
```

Comment:
To sprawdzenie mogłoby się zadziać jeszcze przed haszowaniem. Wtedy jeśli okaże się, że hasła się nie pokrywają to skrypt ma mniej pracy.